evcxr/src/module.rs

```diff
@@ -265,11 +267,47 @@ overflow-checks = true
         )
     }
 }
+fn tee(mut stream: impl Read) -> Result<String, Error> {
+    let mut out: String = String::new();
+    let mut buff: [u8; 1024] = [0; 1024];
+    // I dont like infinite loops
+    loop {
+        let bytes_read = stream.read(&mut buff)?;
+        if bytes_read == 0 {
+            return Ok(out);
+        }
+        let partial_out = std::str::from_utf8(&buff)?;
+        //TODO: maybe pass a callback here
+        print!("{partial_out}");
+        out.push_str(partial_out);
+    }
+}
 
 fn run_cargo(
     mut command: std::process::Command,
     code_block: &CodeBlock,
 ) -> Result<std::process::Output, Error> {
+    let mut child_process = match command
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .spawn()
+    {
+        Ok(child) => child,
+        Err(err) => bail!("Error running 'cargo rustc' as a child process: {}", err),
+    };
+    let child_out = child_process.stdout.take().unwrap();
+    let child_err = child_process.stderr.take().unwrap();
+    let out = match tee(child_out) {
+        Ok(out) => out,
+        Err(err) => bail!("oouch : {}", err),
+    };
+    // TODO: both streams should be processed concurrently.
```

Comment:
It may end up being simplest to spawn a thread to handle one of the streams. Spawning a thread feels like a somewhat heavyweight thing to do, but we're already spawning cargo, so a thread perhaps isn't much in comparison.

Reply:
Hmm makes sense. 
We use thread for now at least.  It's simple. 
It can be changed to some with better performance if needed.